### PR TITLE
Feat login frontend

### PR DESCRIPTION
**Pull request by lindatroesken**
Added MaengelForm and Home. Added ProtectedRoutes and inserted Routes/ProtectedRoutes/Redirect to Form and to Home

### CODE REVIEW

File: frontend/src/pages/Home.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import Page from '../components/Page'
+import Button from '../components/Button'
+import { NavLink } from 'react-router-dom'
+
+export default function Home({ user, ...props }) {
+  return (
+    <Page>
+      <h1>Willkommen {user ? user.username : ''}</h1>
```
Comment (Droggelbecher92) [inline frontend/src/pages/Home.js:8]: Ich mag diese art von if then else :)

File: frontend/src/pages/MaengelForm.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+const initialState = {
+  description: '',
+}
+
+export default function MaengelForm({ user, ...props }) {
+  const [mangel, setMangel] = useState(initialState)
+
+  const handleMangelChange = event => {
+    console.log(event.target.name)
```
Comment (Droggelbecher92) [inline frontend/src/pages/MaengelForm.js:14]: Das ist grad nur übergangsmäßig oder?

**Review by Droggelbecher92 — APPROVED** [review]
Angenehme PR größe :) passt :)

File: frontend/src/pages/MaengelForm.js (hunk 2 above)
Comment (lindatroesken) [reply frontend/src/pages/MaengelForm.js:16]: Ja

File: frontend/src/App.js [hunk 3]
```diff
@@ -25,9 +26,16 @@ export default function App() {
         <Route exact path="/">
           <Home user={user} />
         </Route>
-        <ProtectedRoute user={user} path="/maengel/new">
-          <MaengelForm user={user} />
-        </ProtectedRoute>
```
Comment (lindatroesken) [inline frontend/src/App.js]: Modified routes and added conditional rendering: show only if logged in

File: frontend/src/pages/Home.js [hunk 4]
```diff
@@ -1,21 +1,27 @@
...
-      {user && (
-        <Button>
-          <NavLink to="/maengel/new">Neuer Mangel</NavLink>
-        </Button>
-      )}
-      {!user && (
-        <Button>
-          <NavLink to="/login">Login</NavLink>
-        </Button>
-      )}
```
Comment (lindatroesken) [inline frontend/src/pages/Home.js]: Some changes in Homepage, Header added

File: frontend/src/pages/PersonalMaengelList.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import styled from 'styled-components/macro'
```
Comment (lindatroesken) [inline frontend/src/pages/PersonalMaengelList.js]: This file is new

File: frontend/src/components/Main.js [hunk 6]
```diff
@@ -3,6 +3,9 @@ import styled from 'styled-components/macro'
 export default styled.main`
   display: grid;
   place-items: center;
```
Comment (lindatroesken) [inline frontend/src/components/Main.js:5]: some styling added

File: frontend/src/components/Page.js [hunk 7]
```diff
@@ -2,9 +2,15 @@ import styled from 'styled-components/macro'
 
 export default styled.div`
   position: fixed;
```
Comment (lindatroesken) [inline frontend/src/components/Page.js:4]: some styling added

**Review by AdrianaSche — APPROVED** [review]

**Comment by AdrianaSche** [comment]
Schön!

**Pull request merged**